src/js/angular/similarity/controllers/create-index.controller.js

```diff
@@ -277,6 +280,69 @@ function CreateSimilarityIdxCtrl(
         return $translate.instant('similarity.close.btn.msg', {operation: operationType});
     }
 
+    $scope.getTabErrorMessage = (similarityQueryTab) => {
+
+        if (!$scope.similarityIndexInfo) {
+            return;
+        }
+
+        if (SimilarityQueryType.DATA === similarityQueryTab) {
+           return getSelectQueryErrorMessage();
```

Comment:
Whay not just pass the type and have one method? Build the error key inside based on the type?

Reply:
The method will have to many "if else" operators

Reply:
Well, not sure that you would need more than 2 ifs with 3 branches each.

Reply:
One to check the type and set some properties and the other to check if properties are true and return the correct message.

Reply:
Refactored